### DISCUSSION TITLE
Deployed Keycloak instance using Operator not working?

### DISCUSSION THREAD

**Discussion by bitvijays** [keycloak/keycloak#9179] [Q&A] [answered]
Dear All,

Hope you are doing well and having a good week. Thank you for your support. We are trying to install `keycloak` instance using `keycloak operator` as per [Getting started operator kubernetes](https://www.keycloak.org/getting-started/getting-started-operator-kubernetes). However, it is not getting ready and no pods activity can be viewed. Any support/guidance is appreciated.

We were able to install the operator

```
debian@cloudcore:~$ kubectl get csv -n my-keycloak-operator -o wide
NAME                        DISPLAY             VERSION   REPLACES                    PHASE
keycloak-operator.v15.1.0   Keycloak Operator   15.1.0    keycloak-operator.v15.0.2   Succeeded
```

However, following the next step to install the instance. We executed `kubectl create -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/operator-examples/mykeycloak.yaml` . After a few minutes, Keycloak cluster should be up and running.

However, it is not ready even after 37 mins

```
debian@cloudcore:~$ kubectl get keycloak/mykeycloak -o jsonpath='{.status.ready}'
debian@cloudcore:~$

debian@cloudcore:~$ kubectl get keycloak/mykeycloak
NAME         AGE
mykeycloak   37m
```

`kubectl describe keycloak` also doesn't provide any useful information

```
debian@cloudcore:~$ kubectl describe keycloak/mykeycloak
Name:         mykeycloak
Namespace:    default
Labels:       app=mykeycloak
Annotations:  <none>
API Version:  keycloak.org/v1alpha1
Kind:         Keycloak
Metadata:
  Creation Timestamp:  2021-12-15T14:31:47Z
  Generation:          1
  Managed Fields:
    API Version:  keycloak.org/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
        f:labels:
          .:
          f:app:
      f:spec:
        .:
        f:externalAccess:
          .:
          f:enabled:
        f:instances:
    Manager:         kubectl-client-side-apply
    Operation:       Update
    Time:            2021-12-15T14:31:47Z
  Resource Version:  43109
  UID:               6ceefd3f-8f8b-47b1-bee1-d72eee874445
Spec:
  External Access:
    Enabled:  true
  Instances:  1
Events:       <none>
```

**Comment by andreaTP** [comment, marked as answer]
Hi @bitvijays !
Thanks a lot for reaching out, can you confirm the environment where you are trying to run Keycloak (Openshift /Minikube / ...)?

I suspect that your problem is caused by the fact that the `keycloak-operator` is not watching the `default` namespace where you are installing the CR.
Can you please share the output of this command:

```bash
kubectl exec --tty deployment/keycloak-operator -n my-keycloak-operator -- /usr/bin/env | grep WATCH_NAMESPACE
```

If the environment variable `WATCH_NAMESPACE` is defined than you need to install the CR in the specific namespace, for example:

```bash
kubectl create -f https://raw.githubusercontent.com/keycloak/keycloak-quickstarts/latest/operator-examples/mykeycloak.yaml -n <value-of-WATCH_NAMESPACE>
```

Let me know if this helps.

**Reply by bitvijays** [reply]
@andreaTP Thank you very much! That solves it! Really appreciate the quick response 👍 

Have a minor query, in the documentation [Getting started operator kubernetes](https://www.keycloak.org/getting-started/getting-started-operator-kubernetes), in the section `Login to Keycloak Admin Console with Keycloak Operator` to get the routes, we have mentioned `oc` to find the URLs for the `keycloak`. Shouldn't it be `kubectl` command? 

```
$ KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}')/auth &&
echo "" &&
echo "Keycloak:                 $KEYCLOAK_URL" &&
echo "Keycloak Admin Console:   $KEYCLOAK_URL/admin" &&
echo "Keycloak Account Console: $KEYCLOAK_URL/realms/myrealm/account" &&
echo ""
Keycloak:                 https://keycloak-default.apps.cluster-slaskawi-11a6.slaskawi-11a6.example.opentlc.com/auth
Keycloak Admin Console:   https://keycloak-default.apps.cluster-slaskawi-11a6.slaskawi-11a6.example.opentlc.com/auth/admin
Keycloak Account Console: https://keycloak-default.apps.cluster-slaskawi-11a6.slaskawi-11a6.example.opentlc.com/auth/realms/myrealm/account
```

**Reply by andreaTP** [reply]
@bitvijays the guide you are following covers the installation on OpenShift, please refer to this guide if you are using minikube:

https://www.keycloak.org/getting-started/getting-started-kube

Hope this answer your question!

**Reply by bitvijays** [reply]
Hiya @andreaTP Thank you....I am using Kubernetes on bare metal. But I think...that ip would be the IP of the load balancer. So, it's okay. Thank you for your support ☺️
